dask/dataframe/io.py

```diff
@@ -421,6 +422,65 @@ def _pd_to_hdf(pd_to_hdf, lock, args, kwargs=None):
     return None
 
 
+def _pd_to_bcolz(pd_to_bcolz, lock, args, kwargs=None):
+    if lock:
+        lock.acquire()
+    try:    
+        if kwargs['mode']=='w':
+            pd_to_bcolz(args[0], rootdir=args[1], mode='w') # add expectedlen=outfile_Nrows
+        else: 
+            import_ct = bcolz.ctable(rootdir=args[1], mode='a')
+            temp_ct = bcolz.ctable.fromdataframe(args[0])
+            import_ct.append(temp_ct)
+    finally:
+        if lock:
+            lock.release()
+    
+    return None
+
+
+def to_bcolz(df, outfile):
+    name = 'to-bcolz-' + uuid.uuid1().hex
+    pd_to_bcolz = bcolz.ctable.fromdataframe 
+    
+    kwargs={'mode':'w'}
+    dask_kwargs={}
+    lock=False
+    compute=True
+    name_function=None
+    
+    fmt_obj = lambda outfile, _: outfile
+    if name_function is None: name_function = build_name_function(df.npartitions - 1)
+    i_name = name_function(0)
+
+
+    dsk = dict()
+    dsk[(name, 0)] = (_pd_to_bcolz, pd_to_bcolz, 
+                      lock,
+                      [(df._name, 0), fmt_obj(outfile, i_name)],
+                      kwargs)
+
+
+    kwargs2 = kwargs.copy()
+    kwargs2['mode'] = 'a'
+    for i in range(1, df.npartitions):
+        i_name = name_function(i)
+        task = (_pd_to_bcolz, pd_to_bcolz, lock,
+                [(df._name, i), fmt_obj(outfile, i_name)], kwargs2)
+    
+        link_dep = i - 1
+        task = (_link, (name, link_dep), task)
+        dsk[(name, i)] = task
+
+    dsk = merge(df.dask, dsk)
+    keys = [(name, df.npartitions - 1)]
+
+    if compute:
+        return DataFrame._get(dsk, keys, get=dask.get, **dask_kwargs)
```

Comment:
Calling `threaded.get` on a sequential-reduce pattern like this will likely cause memory blowup.  We often have to switch down to single-threaded get for these cases.  There is a bit of logic that needs to happen to check here.